services/galaxy-toolshed/galaxy-toolshed-version.service.js

```diff
@@ -0,0 +1,163 @@
+import Joi from 'joi'
+import { addv as versionText } from '../text-formatters.js'
+import { version as versionColor } from '../color-formatters.js'
+import { NotFound } from '../index.js'
+import BaseGalaxyToolshedService from './galaxy-toolshed-base.js'
+
+const repositoryRevisionInstallInfoSchema = Joi.array().items(
+  Joi.object({}),
+  Joi.object({
+    changeset_revision: Joi.string().required(),
+    valid_tools: Joi.array()
+      .items(
+        Joi.object({
+          requirements: Joi.array()
+            .items(
+              Joi.object({
+                name: Joi.string().required(),
+                version: Joi.string().required(),
+              }).required()
+            )
+            .required(),
+          id: Joi.string().required(),
+          name: Joi.string().required(),
+          version: Joi.string().required(),
+        }).required()
+      )
+      .required(),
+  }).required(),
+  Joi.object({})
+)
+
+export default class GalaxyToolshedVersion extends BaseGalaxyToolshedService {
+  static category = 'version'
+  static route = {
+    base: 'galaxy-toolshed/v',
+    pattern: ':repositoryName/:owner/:toolId?/:requirementName?',
```

Comment:
This is just too much to try to cram into a single badge, both in terms of the user facing surface of the badge url and in terms of the code. It also ends up accepting invalid paths (e.g. toolId is strictly required for showing the version of a tool's requirement). 

There are 3 very distinct badges here, and we need to reflect our code as such

Reply:
Ok, I will manage to have 3 different routes